README.md
# https-www.pythonanywhere.com-user-gualtierobassetti-consoles-10009407-


history.txt
    1  cd mysite
    2  git status
    3  git add .
    4  git commit -m "<file> -- templates/main_page.html
    5  git add .
    6  git commit -m "<file> -- templates/main_page.html"
    7  mkvirtualenv flask-tutorial --python=python3.6
    8  pip install flask flask-login flask-sqlalchemy mysql-connector-python
    9  pip freeze > requirements.txt
   10  git status
   11  git add .
   12  git commit -m "/.cache/pip/wheels"
   13  pip install Flask-Migrate
   14  export FLASK_APP=flask_app.py
   15  flask db init
   16  flask db migrate
   17  flask db stamp head
   18  pip freeze > requirements.txt
   19  git status
   20  git add .
   21  git commit -m "[alembic.runtime"
   22  flask db migrate
   23  flask db upgrade
   24  git status
   25  git add .
   26  git commit -m "93abb1d37034_.py"
   27  flask db migrate
   28  flask db upgrade
   29  flask shell
   30  flask db migrate
   31  flask db upgrade
   32  flask shell
   33  echo "# https-www.pythonanywhere.com-user-gualtierobassetti-consoles-10009407-" >> README.md
   34  git init
   35  git add README.md
   36  git commit -m "first commit"
   37  git remote add origin https://github.com/mikerissi/https-www.pythonanywhere.com-user-gualtierobassetti-consoles-10009407-.git
   38  git push -u origin master
   39  history
   40  history > history.txt


flask_app.py
from datetime import datetime
from flask import Flask, redirect, render_template, request, url_for
from flask_login import current_user, login_required, login_user, LoginManager, logout_user, UserMixin
from flask_migrate import Migrate
from flask_sqlalchemy import SQLAlchemy
from werkzeug.security import check_password_hash


app = Flask(__name__)
app.config["DEBUG"] = True

SQLALCHEMY_DATABASE_URI = "mysql+mysqlconnector://{username}:{password}@{hostname}/{databasename}".format(
    username="gualtierobassett",
    password="Exeter22",
    hostname="gualtierobassetti.mysql.pythonanywhere-services.com",
    databasename="gualtierobassett$comments",
)
app.config["SQLALCHEMY_DATABASE_URI"] = SQLALCHEMY_DATABASE_URI
app.config["SQLALCHEMY_POOL_RECYCLE"] = 299
app.config["SQLALCHEMY_TRACK_MODIFICATIONS"] = False
db = SQLAlchemy(app)
migrate = Migrate(app, db)

app.secret_key = "5646546546213213213689"
login_manager = LoginManager()
login_manager.init_app(app)


class User(UserMixin, db.Model):

    __tablename__ = "users"

    id = db.Column(db.Integer, primary_key=True)
    username = db.Column(db.String(128))
    password_hash = db.Column(db.String(128))

    def check_password(self, password):
        return check_password_hash(self.password_hash, password)


    def get_id(self):
        return self.username


@login_manager.user_loader
def load_user(user_id):
    return User.query.filter_by(username=user_id).first()


class Comment(db.Model):

    __tablename__ = "comments"

    id = db.Column(db.Integer, primary_key=True)
    content = db.Column(db.String(4096))
    posted = db.Column(db.DateTime, default=datetime.now)
    commenter_id = db.Column(db.Integer, db.ForeignKey('users.id'), nullable=True)
    commenter = db.relationship('User', foreign_keys=commenter_id)


@app.route("/", methods=["GET", "POST"])
def index():
    if request.method == "GET":
        return render_template("main_page.html", comments=Comment.query.all())

    if not current_user.is_authenticated:
        return redirect(url_for('index'))

    comment = Comment(content=request.form["contents"], commenter=current_user)
    db.session.add(comment)
    db.session.commit()
    return redirect(url_for('index'))



@app.route("/login/", methods=["GET", "POST"])
def login():
    if request.method == "GET":
        return render_template("login_page.html", error=False)

    user = load_user(request.form["username"])
    if user is None:
        return render_template("login_page.html", error=True)

    if not user.check_password(request.form["password"]):
        return render_template("login_page.html", error=True)

    login_user(user)
    return redirect(url_for('index'))



@app.route("/logout/")
@login_required
def logout():
    logout_user()
    return redirect(url_for('index'))

migrations/versions/59eaf20f1b83_.py
"""empty message

Revision ID: 59eaf20f1b83
Revises: 
Create Date: 2018-08-29 22:55:48.567939

"""
from alembic import op
import sqlalchemy as sa


# revision identifiers, used by Alembic.
revision = '59eaf20f1b83'
down_revision = None
branch_labels = None
depends_on = None


def upgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    op.create_table('users',
    sa.Column('id', sa.Integer(), nullable=False),
    sa.Column('username', sa.String(length=128), nullable=True),
    sa.Column('password_hash', sa.String(length=128), nullable=True),
    sa.PrimaryKeyConstraint('id')
    )
    op.add_column('comments', sa.Column('commenter_id', sa.Integer(), nullable=True))
    op.add_column('comments', sa.Column('posted', sa.DateTime(), nullable=True))
    op.create_foreign_key(None, 'comments', 'users', ['commenter_id'], ['id'])
    # ### end Alembic commands ###


def downgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    op.drop_constraint(None, 'comments', type_='foreignkey')
    op.drop_column('comments', 'posted')
    op.drop_column('comments', 'commenter_id')
    op.drop_table('users')
    # ### end Alembic commands ###


migrations/versions/93abb1d37034_.py
"""empty message

Revision ID: 93abb1d37034
Revises: 59eaf20f1b83
Create Date: 2018-08-29 23:34:15.190388

"""
from alembic import op
import sqlalchemy as sa


# revision identifiers, used by Alembic.
revision = '93abb1d37034'
down_revision = '59eaf20f1b83'
branch_labels = None
depends_on = None


def upgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    op.create_table('users',
    sa.Column('id', sa.Integer(), nullable=False),
    sa.Column('username', sa.String(length=128), nullable=True),
    sa.Column('password_hash', sa.String(length=128), nullable=True),
    sa.PrimaryKeyConstraint('id')
    )
    op.add_column('comments', sa.Column('commenter_id', sa.Integer(), nullable=True))
    op.add_column('comments', sa.Column('posted', sa.DateTime(), nullable=True))
    op.create_foreign_key(None, 'comments', 'users', ['commenter_id'], ['id'])
    # ### end Alembic commands ###


def downgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    op.drop_constraint(None, 'comments', type_='foreignkey')
    op.drop_column('comments', 'posted')
    op.drop_column('comments', 'commenter_id')
    op.drop_table('users')
    # ### end Alembic commands ###
